core/application/CMakeLists.txt

```diff
@@ -5,15 +5,34 @@ target_link_libraries(application_util
     Boost::filesystem
     )
 
-add_library(app_config_impl
+add_custom_command(
```

Comment:
You could also make a template for this file and configure it with CMake, I think it'd be more readable.